src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -60,7 +60,16 @@ public function registerClasses(Definition $prototype, $namespace, $resource, $e
         $interfaces = array();
         $singlyImplemented = array();
 
-        foreach ($classes as $class) {
+        foreach ($classes as $class => $errorMessage) {
+            if (null !== $errorMessage) {
+                $definition = new Definition($class);
+                $definition->addError($errorMessage);
```

Comment:
We need to think about this error message holistically :). Currently, the user will see:

> Class App\Foo\MissingParentClass not found

I think we need to add more... as much as possible really. Something like:

> Error when loading registering services from resource "../src/*": an error was thrown when processing the class "App\Foo\ActualClass": "Class App\Foo\MissingParentClass not found".

I would LOVE to even say the line number in YAML or XML... but we *definitely* do not know that :p (pipe dream for someday?)

Reply:
👍 This code was more or less meant as a POC. I'll see what I can do about improving the DX.